### PR TITLE
fix workspace persistence issue

### PR DESCRIPTION
**Pull request by imanjra**
## What changes are proposed in this pull request?

Fixes some issues with loading and using workspaces in the app

## How is this patch tested? If it is not, please explain why.

Using app and session

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [x] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

See above

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
  - Introduced automatic clearing of spaces on component mount for improved state management.
  - Added filtering of pending requests for better performance and user experience.

- **Bug Fixes**
  - Updated MongoDB download links for various platforms to ensure compatibility and successful installations.

- **Chores**
  - Organized code by moving `QueueItemStatus` enum to a separate file.
  - Added a build step to upload a wheel artifact for non-sdist platforms.

- **Refactor**
  - Improved state handling and effects in hooks for better performance and reliability.

### CODE REVIEW

**Comment by brimoor** [comment]
@imanjra @benjaminpkane confirmed that workspace persistence works when navigating between datasets via the dataset selector in the App (although I believe the real test bed is Teams here, not OSS).

Similar to my comment https://github.com/voxel51/fiftyone/pull/4469#issuecomment-2156058282, I did notice that if I manually remove `?workspace=WORKSPACE` from the URL bar and press ENTER, the workspace is added back to the URL.

However, unlike for saved views, if I navigate from `http://localhost:5151/datasets/DATASET1?workspace=WORKSPACE` directly to `http://localhost:5151/datasets/DATASET2`, it does correctly load the second dataset without carrying over the workspace.

**Comment by imanjra** [comment]
> Similar to https://github.com/voxel51/fiftyone/pull/4469#issuecomment-2156058282, I did notice that if I manually remove ?workspace=WORKSPACE from the URL bar and press ENTER, the workspace is added back to the URL.

This is intentional for both workspaces and saved views to recover app state based on the current session state. However, changing the workspace/view param in URL to some other workspace/view should trigger session to update along with the app.

**Review by benjaminpkane — APPROVED** [review]

**Comment by imanjra** [comment]
Updated the base to `merge/v0.24.1-to-main`

**Review by benjaminpkane — APPROVED** [review]

File: app/packages/core/src/components/MainSpace/MainSpace.tsx [hunk 1]
```diff
@@ -8,12 +8,17 @@ const { FIFTYONE_SPACE_ID } = constants;
...
+  const { spaces, updateSpaces, clearSpaces } = useSpaces(
+    FIFTYONE_SPACE_ID,
+    sessionSpaces
+  );
   const [panelsState, setPanelsState] = usePanelsState();
   const oldSpaces = useRef(spaces);
   const oldPanelsState = useRef(panelsState);
   const isMounted = useRef(false);
 
+  useEffect(() => clearSpaces, [clearSpaces]);
```
[the remark this answers is not in the document]
Comment (imanjra) [reply app/packages/core/src/components/MainSpace/MainSpace.tsx:20]: clearSpaces is a cleanup method for unmount of the component. The original implementation is intentional

**Pull request merged**